### ISSUE TITLE
IDE0017 is not proposed in top-level code

### ISSUE THREAD

**Issue by DoctorKrolic** [dotnet/roslyn#61066]
**Version Used**:
Latest main (https://github.com/dotnet/roslyn/commit/55bc2b7a1d957238a6cb75682850f27f4584733f)

**Steps to Reproduce**:
```cs
MyClass cl = new(); // 1
cl.MyProperty = 5;

void M()
{
    MyClass cl = new(); // 2
    cl.MyProperty = 5;
}

class MyClass
{
    public int MyProperty { get; set; }
}
```

**Expected Behavior**:
`IDE0017` is reported both in case `// 1` and case `// 2`

**Actual Behavior**:
`IDE0017` is reported only for case `// 2`

**Comment by CyrusNajmabadi** [comment]
We would take a reasonably sized pr for this.

**Comment by DoctorKrolic** [comment]
> a reasonably sized and scoped

You now add this to every issue I open, right 😅

**Comment by CyrusNajmabadi** [comment]
No. This is just the info I add when putting something on the backlog :-).  It's info to anyone interested in taking it on.  :-)
